### ISSUE TITLE
Lingering Worker BLOB

### ISSUE THREAD

**Issue by oze4** [wojtekmaj/react-pdf#452]
- [x] I have read documentation in README
 - [x] I have checked [sample](https://github.com/wojtekmaj/react-pdf/tree/master/sample) and [test](https://github.com/wojtekmaj/react-pdf/tree/master/test) suites to see real life basic implementation
 - [x] I have checked if this question is not already asked
 - [x] I have done a crap ton of research for this issue....

---
#### _I believe I have found a solution for this... I went ahead and created an example/demo that shows how you can dispose of the worker.._

[Demo using `pdf.js`, without `react-pdf`](https://pdfjs.litpoc.com/) to show how to dispose of the worker using the original library - it is rather straight forward.. (`react-pdf` wraps around `pdf.js`)

[Demo using `react-pdf`](https://react-pdf-destroy-worker.litpoc.com/) to show how to dispose of the worker with this library.

[The worker gets created when you run](https://github.com/wojtekmaj/react-pdf/blob/6176ee2b9be5b28a66909029e91b5efec5897b80/src/Document.jsx#L109) `const loadingTask = pdfjs.getDocument(...)` - in order to dispose of the worker you have to run `loadingTask.destroy()`... 

Is there any way to expose the loading task so we can dispose of it if needed? Or expose an event that allows us to dispose of the worker on dismount?

---

**What are you trying to achieve? Please describe.**
Find out why the worker lingers even after the `react-pdf` `<Document/>` component is unmounted.

**Describe what are you trying to achieve Ex. I'd like to display multiple pages of my PDF.**
Free up some RAM, as I'm dealing with pretty large PDF files.

**Describe solutions you've tried**
Manually dispose of the worker - but each time I run: `pdfjs.getDocument(myDocUrl).destroy()` it generates a new `blob:http://url`... which it disposes of immediately after creating it, while still keeping the original worker (the one I'm trying to destroy) around... Is there any way to destroy the worker created by `react-pdf`/`pdfjs`?

**Additional information**
This is how I am grabbing the PDF document from my server:

```jsx
...
    let response = await fetch(fetchUrl, fetchOptions);
    let blob = await response.blob();
    let objectUrl = URL.createObjectURL(blob);
    this.setState({
        file: objectUrl
    });
...
```

**If applicable, add screenshots (preferably with browser console open) and files you have an issue with to help explain your problem.**

From the FireFox Task Manager:
![image](https://user-images.githubusercontent.com/21092343/65057463-432e1400-d938-11e9-9849-bea60ccbd834.png)

This blob url will stay around until i refresh the page... If I browse back to the page which contains the `Document` component, it generates a completely new blob url... This is killing RAM.

**Environment**

 - Firefox 69:
 - React-PDF version: 4.1.0
 - React version: 16.8.6
 - Webpack version (if applicable) [e.g. 4.16.2]: using `create-react-app`

**Comment by oze4** [comment]
***UPDATE:***

---
#### _I believe I have found a solution for this... I went ahead and created an example/demo that shows how you can dispose of the worker.._

[You can find the demo here](https://pdfjs.litpoc.com/) - it is rather straight forward..

[The worker gets created when you run](https://github.com/wojtekmaj/react-pdf/blob/6176ee2b9be5b28a66909029e91b5efec5897b80/src/Document.jsx#L109) `const loadingTask = pdfjs.getDocument(...)` - in order to dispose of the worker you have to run `loadingTask.destroy()`... 

### Is there any way to expose the loading task so we can dispose of it if needed? Or expose an some event that allows us to dispose of the worker on dismount?

---

**Comment by wojtekmaj** [comment]
In other words, every time we load a new document, we should run loadingTask.destroy on the old one? 🤔

**Comment by oze4** [comment]
@wojtekmaj I went ahead and forked the repo and created a demo outlining what I am referring to..

[You can check out the demo here](https://react-pdf-destroy-worker.litpoc.com/) and view the changes I made to the repo can be found [here](https://github.com/oze4/react-pdf/blob/2929888e494f6a8577381e0d6e194fa103a7f253/src/Document.jsx#L110) and [here](https://github.com/oze4/react-pdf/blob/2929888e494f6a8577381e0d6e194fa103a7f253/src/Document.jsx#L79).. These changes resolve this issue..

Please let me know if you would like for me to create a pull request.

**Comment by oze4** [comment]
@wojtekmaj Please let me know if you intend on getting this resolved... If you do not intend on resolving this issue, I will close it as a 'wont-fix'.

**Comment by wojtekmaj** [comment]
Hi, sorry for an embarrassing delay. Yes, could you please go ahead and do a PR with your changes? I'll happily review them.

**Comment by oze4** [comment]
@wojtekmaj  No worries, we all get busy. I went ahead and submitted a PR for this. Please let me know if you have any questions.

Cheers!

**Comment by brentmitch** [comment]
I tried out @oze4 's fix/enhancement and it worked great! When can this be merged?

**Comment by brentmitch** [comment]
I think Issue #504 _There are memory leak phenomenon related to pdf.worker.js_  might be the same issue or at least its related.

**Comment by wojtekmaj** [comment]
Fixed by #505

**Comment by wojtekmaj** [comment]
Note: This fix was cherry-picked to [v4.2.0](https://github.com/wojtekmaj/react-pdf/releases/tag/v4.2.0) release.
